fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -156,31 +157,31 @@ public void commit() {
                 transaction.mutate(valuePair.getLeft(), key, valuePair.getRight()));
         CompletableFuture<byte[]> versionFuture = transaction.getVersionstamp();
         long beforeCommitTimeMillis = System.currentTimeMillis();
-        CompletableFuture<Void> commit = checks.isDone() && !checks.isCompletedExceptionally() ?
+        CompletableFuture<Void> commit = MoreAsyncUtil.isCompletedNormally(checks) ?
                                          transaction.commit() :
                                          checks.thenCompose(vignore -> transaction.commit());
+        commit = commit.thenCompose(vignore -> {
+            // The committed version will be -1 if the transaction is read-only,
+            // in which case versionFuture has completed exceptionally with
+            // transaction_read_only and thus can be ignored.
+            committedVersion = transaction.getCommittedVersion();
+            if (committedVersion > 0) {
+                // The getVersionstamp() future can complete a tiny bit after the commit() future.
+                return versionFuture.thenAccept(vs -> versionStamp = vs);
```

Comment:
Should this use `handle` instead of `thenAccept` and log rather than propagate the error in the case where it can't get the versionstamp? I think that's the current behavior (at least). I'm not sure it buys us much, as the only time it *should* throw an error is if the transaction is actually read-only (which is already being guarded against).

Reply:
If something subtle were happening, I think you'd just get a less clear `NullPointerException` later rather than whatever it is here. I don't see getting the commit version as optional or of secondary importance, really.

Reply:
Okay, makes sense.